.github/workflows/tools.yml

```diff
@@ -0,0 +1,44 @@
+name: Build Tools
+
+on:
+  workflow_dispatch:
+    inputs: 
+      tag:
+        description: 'Tagged commit to build tools against'
+        required: true
+        type: string
+
+jobs:
+  docker-push:
+    name: Push to container registry
+    runs-on: ubuntu-latest
+    steps:
+    - id: auth
+      uses: google-github-actions/auth@v1
+      with:
+        credentials_json: ${{ secrets.GCR_SERVICE_KEY }} # TODO: we need a new key to allow uploads
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.19'
+    - name: Set up Google Cloud SDK
+      uses: google-github-actions/setup-gcloud@v1
+      with:
+        project_id: flow # TODO: is this correct?
+    - name: Checkout repo
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ inputs.tag }}
+    - name: Build relic
+      run: make crypto_setup_gopath
+    - name: Build Tools
+      run: |
+        make tool-bootstrap
+        make tool-util
+        make tool-transit
+    - name: Upload boot-tools
+      run: |
+        mkdir boot-tools
+        mv bootstrap util transit boot-tools/
+        tar -czf boot-tools.tar ./boot-tools/
+        gsutil cp boot-tools.tar gs://flow-genesis-bootstrap/boot-tools.tar
```

Comment:
Could we update this to conditionally support test builds that are uploaded with an artifact name that uses the ref rather than `boot-tools.tar`?

If we also address the comment above, we could essentially use the following to conditionally set the name. This would allow us to kick of on-demand builds that would allow us to build the boot tools for testing purposes.

```
ARTIFACT_NAME=$([ "$INPUT_PRODBUILD" = "1" ] && echo "boot-tools.tar" || echo "$(git rev-parse --short HEAD).tar")
gsutil cp boot-tools.tar gs://flow-genesis-bootstrap/$ARTIFACT_NAME
```

Reply:
how about it uploads to a tag specific directory by default, and there's an option to promote it to the official image
e.g.
* upload to `tools/v0.x.x/boot-tools.tar
* if "promote" is set, also upload to `./boot-tools.tar`